build/tools/Automation/Program.cs

```diff
@@ -2,180 +2,48 @@
-
-        /// <Summary>
-        /// Updates:
-        ///     - constants.yaml
-        ///     - versionsToBuild.txt
-        ///
-        /// Use PlatformConstants to populate constants.yaml and versionsToBuild.txt files
-        /// with relevant new platform release information. The constants.yaml file is populated
-        /// so after build/generateConstants.sh is invoked, the contants.yaml is used to distribute changes
-        /// across Oryx source code. Which allows tests to be automatically to be updated.
-        ///
-        /// <param name="platformConstants">List of PlatformConstant containing platform release information</param>
-        /// <param name="yamlConstants">Deserialized Constants.ConstantsYaml which is ready for editing</param>
-        /// </Summary>
-        public abstract void UpdateConstants(List<PlatformConstant> platformConstants, List<Constant> yamlConstants);
-
-        /// <Summary>
-        /// Stores platform release information so it can be referenced when updating:
-        /// constants.yaml and versionsToBuild.txt files
-        /// </Summary>
-        public class PlatformConstant
-        {
-            /// <Summary>
-            /// The version of the platfom.
-            /// Example: 1.2.3
-            /// </Summary>
-            public string Version { get; set; } = string.Empty;
-
-            /// <Summary>
-            /// The sha of the platform's version.
-            /// Some platforms may not have a sha.
-            /// </Summary>
-            public string Sha { get; set; } = string.Empty;
-
-            /// <Summary>
-            /// The name of the platform.
-            /// Example: dotnet, golang, etc.
-            /// </Summary>
-            public string PlatformName { get; set; } = string.Empty;
-
-            /// <Summary>
-            /// The type of version that is being represented.
-            /// Example: sdk, aspnetcore, netcore, etc.
-            /// </Summary>
-            public string VersionType { get; set; } = string.Empty;
-
-            // TODO: Add fields for other feilds.
-            //  For example, python has GPG keys
-        }
-
-        /// <Summary>
-        /// This is used to deserialize Constants.ConstantsYaml file
-        /// </Summary>
-        public class Constant
-        {
-            public string Name { get; set; } = string.Empty;
-
-            public Dictionary<string, object> Constants { get; set; } = new Dictionary<string, object>();
-
-            public List<object> Outputs { get; set; } = new List<object>();
+            string oryxRootPath = args[1];
+            string platform = args[0].ToLower();
+            var serviceProvider = new ServiceCollection()
+                .AddHttpClient()
+                .AddSingleton<IVersionService, VersionService>()
+                .AddSingleton<IFileService>(new FileService(oryxRootPath))
+                .AddSingleton<IYamlFileService>(new YamlFileService(oryxRootPath))
```

Comment:
nit: would you make sure that the mappings here are in lexicographical order based?

```cs
.AddSingleton<IFileService>(...)
.AddSingleton<IVersionService, VersionService>()
.AddSingleton<IYamlFileService>(...)
```

Reply:
Updated